components/canvas_traits/canvas.rs

```diff
@@ -41,7 +41,8 @@ pub struct CanvasImageData {
 pub enum Canvas2dMsg {
     Arc(Point2D<f32>, f32, f32, f32, bool),
     ArcTo(Point2D<f32>, Point2D<f32>, f32),
-    DrawImage(Option<ByteBuf>, Size2D<f64>, Rect<f64>, Rect<f64>, bool),
+    DrawImage(Size2D<f64>, Rect<f64>, Rect<f64>, bool, IpcSharedMemory),
+    DrawEmptyImage(Size2D<f64>, Rect<f64>, Rect<f64>),
```

Comment:
Would it make more sense to use `Option<IpcSharedMemory>` here instead of adding a new variant?